app/src/main/java/org/mozilla/fenix/tabstray/browser/OtherHeaderBinding.kt

```diff
@@ -19,9 +19,13 @@ class OtherHeaderBinding(
     private val showHeader: (Boolean) -> Unit
 ) : AbstractBinding<TabsTrayState>(store) {
     override suspend fun onState(flow: Flow<TabsTrayState>) {
-        flow.ifAnyChanged { arrayOf(it.normalTabs, it.searchTermGroups) }
+        flow.ifChanged { Pair(it.normalTabs, it.tabPartition) }
             .collect {
-                if (it.searchTermGroups.isNotEmpty() && it.normalTabs.isNotEmpty()) {
+                if (
+                    it.normalTabs.isNotEmpty() &&
+                    !it.tabPartition?.tabGroups?.filter { tabGroup -> tabGroup.tabIds.isNotEmpty() }
+                        .isNullOrEmpty()
```

Comment:
nit: ^^ that would be a nice extension function so we can write `tabPartition.isNotEmpty()`. Let's add that.

Reply:
Sounds good.  I'll add the extension functions. 👍